src/node_report.cc

```diff
@@ -272,20 +272,22 @@ static void WriteNodeReport(Isolate* isolate,
   PrintVersionInformation(&writer);
   writer.json_objectend();
 
-  writer.json_objectstart("javascriptStack");
-  // Report summary JavaScript error stack backtrace
-  PrintJavaScriptErrorStack(&writer, isolate, error, trigger);
+  if (isolate != nullptr) {
+    writer.json_objectstart("javascriptStack");
+    // Report summary JavaScript error stack backtrace
+    PrintJavaScriptErrorStack(&writer, isolate, error, trigger);
 
-  // Report summary JavaScript error properties backtrace
-  PrintJavaScriptErrorProperties(&writer, isolate, error);
-  writer.json_objectend();  // the end of 'javascriptStack'
+    // Report summary JavaScript error properties backtrace
+    PrintJavaScriptErrorProperties(&writer, isolate, error);
+    writer.json_objectend();  // the end of 'javascriptStack'
+
+    // Report V8 Heap and Garbage Collector information
+    PrintGCStatistics(&writer, isolate);
```

Comment:
I guess you moved this call up to better group all the usages of `isolate` right? just thinking aloud here if there is any potential concerns to any consuming tools due to this. I suppose not, because the order of fields in a JSON object does not really matter?

Reply:
Yes, I moved this one up for single branch of isolate existence check. I cannot assume any tools would depends on the order of JSON object keys, also that is not quite practical in most JSON parsers since most likely dictionary like data structure will be generated from JSON objects.